crates/telio-traversal/src/session_keeper.rs

```diff
@@ -391,4 +406,60 @@ mod tests {
         .await
         .unwrap();
     }
+
+    #[tokio::test]
+    async fn test_batcher_invocation() {
+        const PERIOD: Duration = Duration::from_secs(20);
+
+        const THRESHOLD: Duration = Duration::from_secs(10);
+        let socket_pool = Arc::new(SocketPool::new(
+            NativeProtector::new(
+                #[cfg(target_os = "macos")]
+                false,
+            )
+            .unwrap(),
+        ));
+
+        let mut batcher = Box::new(MockBatcherTrait::<telio_crypto::PublicKey, State>::new());
+
+        let pk = "REjdn4zY2TFx2AMujoNGPffo9vDiRDXpGG4jHPtx2AY="
+            .parse::<PublicKey>()
+            .unwrap();
+
+        use mockall::predicate::{always, eq};
+        batcher
+            .expect_add()
+            .once()
+            .with(eq(pk), eq(PERIOD), eq(THRESHOLD), always())
+            .return_once(|_, _, _, _| ());
+        batcher
+            .expect_remove()
+            .once()
+            .with(eq(pk))
+            .return_once(|_| ());
+
+        // it's hard to mock the exact return since it involves a complex type, however we
+        // can at least verify that the batcher's actions were queried
+        batcher
+            .expect_get_actions()
+            .times(..)
+            .returning(|| Err(BatcherError::NoActions));
```

Comment:
use of undeclared type `BatcherError`

Reply:
Thanks for spotting, got left out during refactoring :) Fixed